src/main/java/me/yukun99/ip/core/Ui.java

```diff
@@ -0,0 +1,272 @@
+	}
+
+	/**
+	 * Sends user instructions when requested.
+	 */
+	public void help() {
+		sendMessage(HELP);
+	}
+
+	/**
+	 * Sends user a list of all tasks.
+	 */
+	public void list() {
+		sendMessage(LIST + taskList.toString().replace("\n", NEW_LINE));
+	}
+
+	/**
+	 * Sends user a list of all tasks happening at the specified date in the DateTimePair.
+	 *
+	 * @param pair Date/time pair to send tasks to user for.
+	 */
+	public void listByDate(DateTimePair pair) {
+		sendMessage(LIST_DATE + taskList.listByDate(pair).replace("\n", NEW_LINE));
+	}
+
+	/**
+	 * Sends user a list of all tasks containing the specified word.
+	 *
+	 * @param word Word that has to be in all returned tasks.
+	 */
+	public void findByWord(String word) {
+		sendMessage(FIND + taskFinder.findTasksByWord(word).replace("\n", NEW_LINE));
+	}
+
+	/**
+	 * Sends user a message telling them task has been added.
+	 *
+	 * @param task Task that has been added.
+	 */
+	public void add(Task task) {
+		sendMessage(ADD
+				+ NEW_LINE + task
+				+ NEW_LINE + remaining());
+	}
+
+	/**
+	 * Sends user a message telling them task is already done.
+	 *
+	 * @param task Task that has already been done.
+	 */
+	public void alreadyDone(Task task) {
+		sendMessage(DONE_COMPLETED
+				+ NEW_LINE + task);
+	}
+
+	/**
+	 * Sends user a message telling them a task has been marked as done.
+	 *
+	 * @param task Task that has been marked as done.
+	 */
+	public void done(Task task) {
+		sendMessage(DONE_UNCOMPLETED
+				+ NEW_LINE + task);
+	}
+
+	/**
+	 * Gets message containing information on number of tasks in the list.
+	 *
+	 * @return Message containing information on number of tasks in the list.
+	 */
+	private String remaining() {
+		return REMAINING.replace(TASKNUM_PLACEHOLDER, taskList.getRemaining() + "");
+	}
+
+	/**
+	 * Sends user a message telling them a task has been updated.
+	 *
+	 * @param task Task that has been updated.
+	 */
+	public void update(Task task) {
+		sendMessage(UPDATE + NEW_LINE + task);
+	}
+
+	/**
+	 * Sends user a message telling them a task has been deleted.
+	 *
+	 * @param task Task that has been deleted.
+	 * @param done Whether deleted task has been done.
+	 */
+	public void delete(Task task, boolean done) {
+		String reply;
+		if (done) {
+			reply = DELETE_DONE;
+		} else {
+			reply = DELETE_UNDONE;
+		}
+		reply += NEW_LINE + task
+				+ NEW_LINE + remaining();
+		sendMessage(reply);
+	}
+
+	/**
+	 * Sends user exit message.
+	 */
+	public void exit() {
+		sendMessage(EXIT);
+	}
+
+	/**
+	 * Sends user error message.
+	 *
+	 * @param e Error to be included in the message.
+	 */
+	public void error(Exception e) {
+		sendMessage(e.toString());
+	}
+
+	/**
+	 * Sends formatted messages to the user.
+	 *
+	 * @param message Message to be formatted.
+	 */
+	private void sendMessage(String message) {
+		String reply = REPLY_HEADER
+				+ NEW_LINE + message
+				+ NEW_LINE + REPLY_FOOTER
+				+ NEW_LINE;
```

Comment:
Should wrapped lines be indented 8 spaces after the parent line?

Reply:
Yes, refer to [here](https://se-education.org/guides/conventions/java/intermediate.html#layout) for the related convention!